providers/github/connection/connection.go

```diff
@@ -161,5 +165,73 @@ func newGithubTokenClient(conf *inventory.Config) (*github.Client, error) {
 		&oauth2.Token{AccessToken: token},
 	)
 	tc := oauth2.NewClient(ctx, ts)
-	return github.NewClient(tc), nil
+	return github.NewClient(newGithubRetryableClient(tc)), nil
+}
+
+func newGithubRetryableClient(httpClient *http.Client) *http.Client {
+	retryClient := retryablehttp.NewClient()
+	retryClient.RetryMax = 5
+	retryClient.Logger = &zeroLogAdapter{}
+
+	if httpClient == nil {
+		httpClient = http.DefaultClient
+	}
+	retryClient.HTTPClient = httpClient
+
+	retryClient.CheckRetry = func(ctx context.Context, resp *http.Response, err error) (bool, error) {
+		// Default Retry Policy would not retry on 403 (adding 429 for good measure)
+		if resp.StatusCode == 403 || resp.StatusCode == 429 {
+			// Primary and Seconday rate limit
+			if resp.Header.Get("x-ratelimit-remaining") == "0" {
+				return true, nil // Should be retried after the rate limit reset (duration handled by Backoff)
+			}
+		}
+		return retryablehttp.DefaultRetryPolicy(ctx, resp, err)
+	}
+	retryClient.Backoff = func(min, max time.Duration, attemptNum int, resp *http.Response) time.Duration {
+		if resp.StatusCode == 403 || resp.StatusCode == 429 {
+			// Primary and Seconday rate limit
+			if resp.Header.Get("x-ratelimit-remaining") == "0" {
+				unix, err := strconv.ParseInt(resp.Header.Get("x-ratelimit-reset"), 10, 64) // x-ratelimit-reset	- The time at which the current rate limit window resets, in UTC epoch seconds
+				if err != nil {                                                             // Must be impossible to hit errors here, but just in case
+					return time.Second * 8 // Github starts with 8s jailtime
```

Comment:
Where did you find this? I only found a one minute pause on the secondary limit.

Do we need to handle primary and secondary separately?

Reply:
Couldn't find at this point.
What would you suggest for error handling?

I don't think so, no, it'll be reflected in wait time. But, I think we could add the maximum wait time and throw error after that, because otherwise the scan can be stuck up to an hour and provide no data

Reply:
Ok, I understood your comment in a way that it is some default GitHub setting.
The logs do not explicitly mention that they use a sliding window for rate limits, but I hope so. 8s should then be fine.

I like the idea with the max wait time.